invokeai/app/invocations/generate.py

```diff
@@ -41,18 +45,32 @@ class TextToImageInvocation(BaseInvocation):
 
     # TODO: pass this an emitter method or something? or a session for dispatching?
     def dispatch_progress(
-        self, context: InvocationContext, sample: Any = None, step: int = 0
-    ) -> None:
+        self, context: InvocationContext, sample: Tensor, step: int
+    ) -> None:  
+        # TODO: only output a preview image when requested
+        image = Generator.sample_to_lowres_estimated_image(sample)
```

Comment:
I thought we were trying to get rid of any references to `Generator`? Can this be moved elsewhere?

Reply:
This `Generator` is the base class for `Txt2Img` which is being used in this same file.

Really, this should probably be a util, I think it was tacked on to `Generator` for organizational purposes. I'm not sure if anything else relies on it being a method on that class.

Reply:
There are no other references to this method anywhere, so moving it seems fine. Regardless, I'm going to leave it where it is for now